build-info-extractor/src/main/java/org/jfrog/build/extractor/executor/CommandExecutor.java

```diff
@@ -204,7 +194,37 @@ private static Process runProcess(File execDir, List<String> args, List<String>
             }};
         }
         logCommand(logger, args, credentials);
-        return Runtime.getRuntime().exec(args.toArray(new String[0]), env, execDir);
+        args = args.stream().map(CommandExecutor::unQuote).collect(Collectors.toList());
+        ProcessBuilder processBuilder = new ProcessBuilder(args)
+                .directory(execDir);
+        processBuilder.environment().putAll(env);
+        return processBuilder.start();
+    }
+
+    /**
+     * Escape spaces in the input executable path and trim leading and trailing whitespaces.
+     *
+     * @param executablePath - the executable path to process
+     * @return escaped and trimmed executable path.
+     */
+    private static String escapeSpacesInPath(String executablePath) {
+        if (executablePath == null) {
+            return null;
+        }
+        return executablePath.trim().replaceAll(" ", SystemUtils.IS_OS_WINDOWS ? "^ " : "\\\\ ");
+    }
+
+    /**
+     * Removes the surrounding quotes from the given string, if present.
+     *
+     * @param str The string to unquote.
+     * @return The unquoted string.
+     */
+    public static String unQuote(String str) {
```

Comment:
The word "unquote" has a meaning different from the one used here, so I suggest renaming this method.
Suggestion: `removeQuotes`